### PR TITLE
[allocation] prefetch memory into caches that just got allocated for an object

### PR DESCRIPTION
**Pull request by lewurm**
the idea is that allocation is a hard to predict access pattern for the CPU. not sure if it's worth exploring.

### CODE REVIEW

**Comment by lewurm** [comment]
builddeb

File: mono/mini/mini-amd64.c [hunk 1]
```diff
@@ -5686,6 +5686,9 @@ mono_arch_output_basic_block (MonoCompile *cfg, MonoBasicBlock *bb)
 				x86_mfence (code);
 			break;
 		}
+		case OP_PREFETCH:
```
Comment (alexrp) [inline mono/mini/mini-amd64.c]: Don't forget to at least add an empty case to other backends.

**Comment by kumpera** [comment]
Does it help at all perf?
TLAB is usually cache hot.

**Comment by lewurm** [comment]
it's too much noise on my machine, so I wanted to see what the benchmark machines show.  I'm afraid it's nothing :/

**Comment by lewurm** [comment]
build deb

**Comment by lewurm** [comment]
build deb

**Comment by lewurm** [comment]
build deb

**Comment by lewurm** [comment]
benchmark results are not convincing.

**Pull request closed without merging**